dev/win32 minor fixes

zerhacken opened the pull request:
- fix uv_handle_size and uv_req_size when failing return 0
- fix VC++ warning C4244
- fix VC++ warning C4389

zerhacken committed:
src: VC++ fix warning C4244

zerhacken committed:
src: fix VC++ warning C4244

Nodejs-Jenkins commented:
Thank you for contributing this pull request! Here are a few pointers to make sure your submission will be considered for inclusion.

Commit zerhacken/libuv@77a7ef6f6602089728f48218149ef662a3222977 has the following error(s):
- First line of commit message must be no longer than 50 characters

The following commiters were not found in the CLA:
- Rasmus Christian Pedersen

You can fix all these things _without_ opening another issue.

Please see [CONTRIBUTING.md](https://github.com/joyent/node/blob/master/CONTRIBUTING.md) for more information

zerhacken commented:
@Nodejs-Jenkins signed CLA

retep998 commented on src/uv-common.c:
```diff
@@ -52,7 +52,7 @@ size_t uv_handle_size(uv_handle_type type) {
     UV_HANDLE_TYPE_MAP(XX)
     default:
-      return -1;
+      return 0;
```
Wouldn't it be better to return (size_t)-1 here?

zerhacken commented:
@retep998 thanks, you are right. Will fix it.

zerhacken committed:
src: fix uv_handle_size uv_req_size

Change uv_handle_size and uv_req_size to return 0
when failing to detect size. C99 defines size_t as an unsigned
integer type. This commit also fixes VC++ warning C4245

zerhacken committed:
src: fix VC++ warning C4389

saghul commented on src/uv-common.c:
```diff
@@ -52,7 +52,7 @@ size_t uv_handle_size(uv_handle_type type) {
     UV_HANDLE_TYPE_MAP(XX)
     default:
-      return -1;
+      return (size_t) - 1;
```
This is not a good idea, it will return the biggest possible integer, which is not what we want.

zerhacken commented:
Revert to my original fix returning 0?

saghul commented:
I guess so, yes. The size of a handle needs to be > 0.

zerhacken commented:
@saghul reverted to original fix, hope this is better?

saghul commented on src/uv-common.c:
```diff
@@ -101,7 +101,7 @@ const char* uv_strerror(int err) {
   memset(addr, 0, sizeof(*addr));
   addr->sin_family = AF_INET;
-  addr->sin_port = htons(port);
+  addr->sin_port = htons((u_short) port);
```
I'm not seeing any warning here, and u_short is a Windows-ism. uint16_t would be fine but I've never seen a warning, what system are you using?

zerhacken replied:
VC++ (2012 + 2013) its a /L4 warning, you are right uint16_t would be better.

saghul replied:
What is a /L4 warning?

Also, the function itself accepts an int, but ports cannot be negative, so maybe we need to adjust uv_ip4_add itself...

zerhacken replied:
/L4 short hand for compiler warning level 4 (http://msdn.microsoft.com/en-us/library/thxezb7y.aspx).

Will look into adjusting uv_ip4_add to take an unsigned int if you are fine with that?
Actually "uv_ip4_add" might as well take an unsigned short?

saghul commented on src/uv-common.c:
```diff
@@ -115,7 +115,7 @@ int uv_ip6_addr(const char* ip, int port, struct sockaddr_in6* addr) {
   memset(addr, 0, sizeof(*addr));
   addr->sin6_family = AF_INET6;
-  addr->sin6_port = htons(port);
+  addr->sin6_port = htons((u_short) port);
```
Ditto.

saghul commented on src/win/poll.c:
```diff
@@ -209,7 +209,7 @@ static int uv__fast_poll_set(uv_loop_t* loop, uv_poll_t* handle, int events) {
   assert((events & ~(UV_READABLE | UV_WRITABLE)) == 0);
 
-  handle->events = events;
+  handle->events = (unsigned char) events;
```
Maybe we should declare events as unsigned int instead.

zerhacken replied:
That would be perfect.

saghul commented on src/win/poll.c:
```diff
@@ -451,7 +451,7 @@ static int uv__slow_poll_set(uv_loop_t* loop, uv_poll_t* handle, int events) {
   assert((events & ~(UV_READABLE | UV_WRITABLE)) == 0);
 
-  handle->events = events;
+  handle->events = (unsigned char) events;
```
Ditto.

zerhacken committed:
src: minor API change uv_ipx_addr and uv_poll_start

* Change uv_ip4_addr and uv_ip6_addr to take an unsigned short as
  port number instead of int(signed).
* Change uv_poll_start events to unsigned int instead of an
  int(signed). This also changes uv_poll_t.

zerhacken committed:
Merge branch 'dev/win32-minor-fixes' of https://github.com/zerhacken/libuv into dev/win32-minor-fixes

zerhacken commented:
@saghul this I like better, should be warning-less and even a tad prettier

zerhacken committed:
src: minor API change uv_poll_start

* Change uv_poll_start pevents to unsigned int instead of an
  int(signed).

zerhacken commented:
The test failure on smartos build doesnt appear as related
